src/Cart/CartCondition.php

```diff
@@ -158,6 +176,10 @@ public function onLoad()
     {
     }
 
+    public function getModel()
+    {
+    }
+
```

Comment:
I don't think we need to define this here as not all cartcondition will require it

Reply:
This code is required here even for all conditions other than `Coupon`. Removing this code would result into error as below.

        
![image](https://github.com/tastyigniter/flame/assets/20902296/9690a7d5-abbc-4219-b18b-c9b01137f6d4)

Reply:
Looking at the code, `getModel` method is only available on the coupon cartcondition, same as `appliesOnMenuItems` and `getApplicableItems` methods. I think we should move the logic into the coupon condition since it is more specific to the coupon and not all cartconditions would have a getmodel or applicable items method